### DISCUSSION TITLE
How do I cancel a subscription

### DISCUSSION THREAD

**Discussion by javaHBZ** [eclipse-milo/milo#905] [General]
I found a class called 'CloseSessionRequest' but wondered if it could be resolved

**Comment by kevinherron** [comment]
If you are using the `ManagedSubscription` abstraction then call `ManagedSubscription::delete`.

Otherwise, you can use `UaSubscriptionManager::deleteSubscription`.

**Reply by javaHBZ** [reply]
Thank you for your answer. The problem has been solved according to your explanation. Thank you very much
